confd/etc/calico/confd/templates/bird.cfg.template

```diff
@@ -42,13 +42,27 @@ router id {{if eq "hash" ($router_id) -}}
 	{{if ne "" ($router_id)}}{{$router_id}}{{else}}{{$node_ip}}{{end}};
 {{- end}}
 
+{{- $bind_mode := ""}}
+{{- $bind_mode_key := printf "/host/%s/bind_mode" (getenv "NODENAME")}}
```

Comment:
Ah, I think the problem is that the client.go code is only handling `BindMode` in the global BGPConfiguration (i.e. the BGPConfiguration with name "default"), but here the template is checking for a per-node `bind_mode` value.

`BindMode` in global BGPConfiguration -> template path "/global/bind_mode".

`BindMode` in per-node BGPConfiguration for node XYZ -> template path "/host/XYZ/bind_mode".

I think you want global, so please try updating this template coding to use "/global/bind_mode".

Reply:
I have modified the template coding to use "/global/bind_mode". Inspite of that , i see that its not being set in bird.cfg.template. I put debug logs in both `getBindModeKVPair `and `GetValues `, i see that value is being set properly. I am pasting logs from GetValues method.

We can see `/calico/bgp/v1/global/bind_mode:NodeIP`

`2022-01-21 07:15:10.515 [INFO][90] confd/client.go 1421: Returning (map[/calico/bgp/v1/global/as_num:64512 /calico/bgp/v1/global/bind_mode:NodeIP /calico/bgp/v1/global/listen_port:176 /calico/bgp/v1/global/loglevel:info /calico/bgp/v1/global/node_mesh:{"enabled": true} ...... ]) results`